src/caffe/util/io.cpp

```diff
@@ -32,35 +33,35 @@ using google::protobuf::io::CodedOutputStream;
 using google::protobuf::Message;
 
 bool ReadProtoFromTextFile(const char* filename, Message* proto) {
-  int fd = open(filename, O_RDONLY);
-  CHECK_NE(fd, -1) << "File not found: " << filename;
-  FileInputStream* input = new FileInputStream(fd);
+  FILE* fp = fopen(filename, "r");
+  CHECK_NOTNULL(fp);
+  FileInputStream* input = new FileInputStream(fileno(fp));
   bool success = google::protobuf::TextFormat::Parse(input, proto);
   delete input;
-  close(fd);
+  fclose(fp);
   return success;
 }
 
 void WriteProtoToTextFile(const Message& proto, const char* filename) {
-  int fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC, 0644);
-  FileOutputStream* output = new FileOutputStream(fd);
+  FILE* fp = fopen(filename, "w");
+  FileOutputStream* output = new FileOutputStream(fileno(fp));
   CHECK(google::protobuf::TextFormat::Print(proto, output));
   delete output;
-  close(fd);
+  fclose(fp);
 }
 
 bool ReadProtoFromBinaryFile(const char* filename, Message* proto) {
-  int fd = open(filename, O_RDONLY);
-  CHECK_NE(fd, -1) << "File not found: " << filename;
-  ZeroCopyInputStream* raw_input = new FileInputStream(fd);
+  FILE* fp = fopen(filename, "r");
+  CHECK_NOTNULL(fp);
+  ZeroCopyInputStream* raw_input = new FileInputStream(fileno(fp));
   CodedInputStream* coded_input = new CodedInputStream(raw_input);
```

Comment:
CodedInputStream already handles buffering:

```
If you want to be able to read the old-fashion way, you can create a CodedInputStream or CodedOutputStream wrapping these objects and use their ReadRaw()/WriteRaw() methods. These will, of course, add a copy step, but Coded*Stream will handle buffering so at least it will be reasonably efficient.
```

so I don't see the use of this diff. As @seanbell  points out, you're creating a buffered abstraction, pulling out the fd anyway, then passing it to another buffering abstraction, whereas the previous behavior was just to pass the fd to the buffering abstraction.